### PR TITLE
fix: Add lock around dbt cache manipulation

### PR DESCRIPTION
**Pull request by chamini2**

### CODE REVIEW

**Comment by mederka** [comment]
Looks like a test has failed. Is there are race condition?

**Comment by chamini2** [comment]
Ran it 12 times: 
image

It seems we are good.

File: src/faldbt/lib.py [hunk 1]
```diff
@@ -101,6 +104,38 @@ def _get_adapter(
...
+        yield
+    except BaseException as exc:
+        _exc = exc
+
+    _lock.release()
+    logger.debug("Released {}", operationId)
+    if _exc:
+        raise RuntimeError(
+            f"Error during cache lock (operation: {operationId})"
+        ) from _exc
```
Comment (isidentical) [inline src/faldbt/lib.py:131]: Let's maybe just use try/finally here and propagate the exception directly (unless there is a reason to stick with `RuntimeError`, which is a behavioral change),

```py
    try:
        yield
    finally:
        _lock.release()
        logger.debug("Released {}", operationId)
```

**Review by isidentical — APPROVED** [review]

File: src/faldbt/lib.py (hunk 1 above)
Comment (chamini2) [reply src/faldbt/lib.py]: But it does work. Thanks!

**Pull request merged**